cpp/arcticdb/version/python_bindings.cpp

```diff
@@ -405,55 +406,55 @@ void register_bindings(py::module &version, py::exception<arcticdb::ArcticExcept
              py::arg("library"),
              py::arg("license_key") = std::nullopt)
         .def("write_partitioned_dataframe",
-             &PythonVersionStore::write_partitioned_dataframe,
-             "Write a dataframe to the store")
+             &PythonVersionStore::write_partitioned_dataframe, 
+             py::call_guard<SingleThreadMutexHolder>(), "Write a dataframe to the store")
```

Comment:
I have thought about that but the `call_guard` is only needed for functions involves folly::future, which are not the majority of the case. So at the end of the day, developers still need to differentiate whether they need the guard or not. Then, in your case, assigning the new subclass of `py::class_`. To me, it is not a big different to adding the `py::call_guard`.
Or, if you agree adding the mutex to add functions, it will justify this idea, I think.

Reply:
I was thinking just for `PythonVersionStore`, but I can see that my idea might add about as much complexity as it removes. Let's leave it as-is for now.